### DISCUSSION TITLE
Recycle bin. Show only open and empty

### DISCUSSION THREAD

**Discussion by quanticplus04** [moudey/Shell#489] [Q&A] [answered]
How to show only these two options when recycle bin is selected ?

**Comment by quanticplus04** [comment]
So far so good…I could remove SOME of the entries. Menus like "Terminal" won't remove neither personal entries (here "laragon"). Any clues ?
//show only open and empty for recycle bin
modify(type="recyclebin" where=window.is_desktop find="Épingler|renommer|Propriétés|Terminal|laragon" vis="hidden")
BEFORE:
![image](https://github.com/user-attachments/assets/c01916e9-2d3d-4376-97cf-f7ee10fe804f)
AFTER:
![image](https://github.com/user-attachments/assets/349410dd-c74e-4b5e-9752-e6a3c156b02a)

**Comment by quanticplus04** [comment]
Trying NOR …same result
modify(type="recyclebin" where=window.is_desktop find='!"Ouvrir"' vis="hidden")
(How to exclude all BUT "Open" and "Empty" ? )
![image](https://github.com/user-attachments/assets/6dfa4039-4e42-4b7d-ad84-4f3b80549f9a)

**Comment by moudey** [comment]
```js
modify(type="recyclebin" where=window.is_desktop and !(this.name=="open" or this.name.start("empty")) vis=0)
```
or by regex
```js
modify(type="recyclebin" where=window.is_desktop and !regex.match(this.name,'(^open$|^empty.*)') vis=0)
```

You must manually exclude new items from the Recycle Bin if you do not need them to appear in the context menu

**Reply by quanticplus04** [reply]
Ty ! Using french, the second option work with
//show only open and empty for recycle bin

modify(type="recyclebin" where=window.is_desktop and !regex.match(this.name,'(^ouvrir$|^vider.*)') vis=0)

![image](https://github.com/user-attachments/assets/9503501a-59ca-46dc-8011-2af38ae12806)

**Reply by Eta-Beta** [reply]
> or by regex

Good morning, applying your second code this happens, also I don't understand why open is at the bottom and empty trash at the top. 

![Screenshot 2025-03-29 182826](https://github.com/user-attachments/assets/8b21c54a-11ae-416a-b64b-8aa9b66089b3)

**Reply by quanticplus04** [reply]
Funny. Try moving "apri" on top with something like
modify ( type"recyclebin" find="Apri" pos=top)

**Reply by Eta-Beta** [reply]
Thanks a lot for the reply, I solved almost everything with the help of "whipper74", in this thread [https://github.com/moudey/Shell/discussions/658#discussion-8132495](url) , the procedure is different but it works, except for "More Options", I couldn't remove it in any way. If you have any ideas let me know.

**Comment by ghost** [comment, marked as answer]
The remaining entries in your case are those that are added by `Nilesoft Shell`, `modify/remove` work for system items or those added by 3rd party *only*. So you have to manipulate remaining items with Shell code. For removing `more options` add `type='~Recyclebin|~Taskbar'` inside `shell.nss` like:
```js
menu(mode="multiple" title=title.more_options image=icon.more_options type='~Recyclebin|~Taskbar' )
{
}
```
Similarly, you can remove other items.

**Reply by RubicBG** [reply]
https://github.com/moudey/Shell/discussions/658#discussioncomment-12680368

**Comment by quanticplus04** [comment]
Ty Xhib-ba. It work.  I understood your line (newbie here) and managed to apply it to the various menus created by Shell (Terminal, Apps, so on…)
![image](https://github.com/user-attachments/assets/63c76136-ea21-4d64-856a-82cf2573592a)

**Comment by whipper74** [comment]
I wish you could explain the issue of removing excess options from the recycle bin in a clear manner.

Friends who ask questions here should also write the solutions they find here in an understandable way so that other people can benefit from them.

It is not easy for me to explain my problem in English, I apologize for the mistakes.

**Comment by quanticplus04** [comment]
Hi Whipper74. Here is the line I added in my perso.nss

//show only open and empty for recycle bin

modify(type="recyclebin" where=window.is_desktop and !regex.match(this.name,'(^ouvrir$|^vider.*)') vis=0)

Explanation
Shell will modify options in recyclebin and set invisible options others than open and empty… (ouvrir and vider in french)

This following line wasn't necessary in my case so I commented it.
//menu(mode="multiple" title=title.more_options image=icon.more_options type='~Recyclebin' vis=0 )
//{
//}

Hope it help

**Reply by whipper74** [reply]
Thank you for taking the time to answer.

Great.

**Reply by whipper74** [reply]
> Hi Whipper74. Here is the line I added in my perso.nss
> 
> //show only open and empty for recycle bin
> 
> modify(type="recyclebin" where=window.is_desktop and !regex.match(this.name,'(^ouvrir$|^vider.*)') vis=0)
> 
> Explanation Shell will modify options in recyclebin and set invisible options others than open and empty… (ouvrir and vider in french)
> 
> This following line wasn't necessary in my case so I commented it. //menu(mode="multiple" title=title.more_options image=icon.more_options type='~Recyclebin' vis=0 ) //{ //}
> 
> Hope it help


Hello,
I have recently discovered the Shell application. I can't say I am very good at software and coding, but I’m not completely inexperienced either.

I have a problem with the Shell application, and despite all my attempts and coding, I haven’t been able to succeed.

The issue has been briefly discussed here, but even after all my attempts, I couldn’t make it work.

To summarize, although I can easily hide Windows' options from the Recycle Bin right-click menu, I am unable to block the menu options added by the Shell application. For example, in the Recycle Bin right-click menu, there is a terminal menu created by the Shell application. Even if I want to remove, delete, or hide it somehow, none of the methods I tried worked. I didn't achieve success with any of the methods, including the code mentioned here.

How can this be done? Is there a way to do this? You mentioned that you succeeded, but when I try the code you added, the options "Open" and "Empty Recycle Bin" that I want to remain in the Recycle Bin right-click menu are removed, and the options I want to remove stay. I want exactly the opposite. Is there a simple way to do this?

I apologize for my bad English, and thank you.

**Comment by whipper74** [comment]
In short, I want only "Open" and "Empty Recycle Bin" to remain in the Recycle Bin, and the rest to be removed

**Comment by quanticplus04** [comment]
Funny ! Could you post your code ? shell.nss in case you modified it (In my case, I just mentioned the import )and your import file. In my case import 'imports/perso.nss'. The only line I modify for recycle bin was :

//show only open and empty for recycle bin
modify(type="recyclebin" where=window.is_desktop and !regex.match(this.name,'(^ouvrir$|^vider.*)') vis=0)

Voilà.

**Reply by whipper74** [reply]
Hello, what is the perso.nss you mentioned here? I don't have an NSS file with this name. Did you create this NSS file yourself? I didn't understand.

**Comment by quanticplus04** [comment]
Yes. To put my own modifications
